pymc_marketing/mmm/transformers.py

```diff
@@ -715,3 +715,58 @@ def tanh_saturation_baselined(
     Developed by Max Kochurov and Aziz Al-Maeeni doing innovative work in `PyMC Labs <pymc-labs.com>`_.
     """
     return gain * x0 * pt.tanh(x * pt.arctanh(r) / x0) / r
+
+
+def michaelis_menten(
+    x: Union[float, np.ndarray, npt.NDArray[np.float64]],
+    alpha: Union[float, np.ndarray, npt.NDArray[np.float64]],
+    lam: Union[float, np.ndarray, npt.NDArray[np.float64]],
+) -> Union[float, Any]:
+    """
+    Evaluate the Michaelis-Menten function for given values of x, alpha, and lambda.
+
+    The Michaelis-Menten function is a type of mathematical saturation function commonly used in
+    enzyme kinetics, but it's also applicable in marketing mix models to describe
+    how different channels contribute to a certain outcome (e.g., sales or conversions)
+    as the spending on that channel increases and the contribution saturates.
+
+    Mathematically, it is described as:
+    α * x / (λ + x)
+
+    Parameters
+    ----------
+    x : float
+        The spent on a channel.
+    alpha (Limit/Vmax) : float
+        The maximum contribution a channel can make.
+    lam (k) : float
+        The elbow on the function in `x` (Point where the curve change their direction).
+
+    Returns
+    -------
+    float
+        The value of the Michaelis-Menten function given the parameters.
+    """
+
+    return alpha * x / (lam + x)
+
+
+def sigmoid_saturation(
+    x: Union[float, np.ndarray, npt.NDArray[np.float64]],
+    alpha: Union[float, np.ndarray, npt.NDArray[np.float64]],
+    lam: Union[float, np.ndarray, npt.NDArray[np.float64]],
+) -> Union[float, Any]:
+    """
+    Parameters
+    ----------
+    - alpha
+        α (alpha): Represent the Asymptotic Maximum or Ceiling Value.
+    - lam
+        λ (lambda): affects how quickly the function approaches its upper and lower asymptotes. A higher value of
+        lam makes the curve steeper, while a lower value makes it more gradual.
+    """
+
+    if alpha <= 0 or lam <= 0:
+        raise ValueError("alpha and lam must be greater than 0")
```

Comment:
I believe this raises a TypeError if these are `TensorVariable`
Are these checks required for the optimization?

Reply:
i.e. 
```python
import pytensor.tensor as pt

if pt.as_tensor_variable([0, 1, 2]) <= 0: # Causes issue
    print("HERE")